config/config.go

```diff
@@ -489,6 +477,24 @@ func (c *Config) ValidateGlobalConfig(ctx context.Context) *gerr.GatewayDError {
 		seenConfigObjects = append(seenConfigObjects, "servers")
 	}
 
+	// Each server configuration should have at least one proxy defined.
+	// Each proxy in the server configuration should be referenced in proxies configuration.
+	for serverName, server := range globalConfig.Servers {
+		if len(server.Proxies) == 0 {
+			err := fmt.Errorf("\"servers.%s\"  has no proxies defined", serverName)
```

Comment:
I know such code was already present, and you only copy-pasted + adapted,  but please consider doing this

```suggestion
			err := fmt.Errorf(`"servers.%s"  has no proxies defined`, serverName)
```